core/src/main/java/io/questdb/cutlass/http/processors/LineHttpProcessor.java

```diff
@@ -146,6 +147,9 @@ public void onRequestComplete(HttpConnectionContext context) throws PeerDisconne
         if (state.isOk()) {
             state.setSendStatus(SendStatus.HEADER);
             context.simpleResponse().sendStatusNoContent(204);
+            Metrics metrics = engine.getMetrics();
+            long totalReceivedHttpBytes = metrics.line().totalHttpBytesGauge().getValue();
+            metrics.line().totalHttpBytesGauge().setValue(totalReceivedHttpBytes + context.getTotalReceived());
```

Comment:
Please use `LongGauge.add()` to avoid races
```suggestion
            metrics.line().totalHttpBytesGauge().add(context.getTotalReceived());
```

Reply:
It would make sense to increment bytes received even in failures, e.g. move this outside of the `if (state.isOk()) / else`